x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -0,0 +1,329 @@
+import { callApmApi } from '../../../../services/rest/createCallApmApi';
+import { TransactionDetailLink } from '../../../shared/Links/apm/TransactionDetailLink';
+import { TransactionOverviewLink } from '../../../shared/Links/apm/TransactionOverviewLink';
+import { TableFetchWrapper } from '../../../shared/table_fetch_wrapper';
+import { TableLinkFlexItem } from '../table_link_flex_item';
+import { NOT_AVAILABLE_LABEL } from '../../../../../common/i18n';
+import { SparkPlotWithValueLabel } from '../../../shared/charts/spark_plot/spark_plot_with_value_label';
+import { ImpactBar } from '../../../shared/ImpactBar';
+
+interface Props {
+  serviceName: string;
+}
+
+type SortField = 'latency' | 'traffic' | 'error_rate' | 'impact';
+type SortDirection = 'asc' | 'desc';
+
+const PAGE_SIZE = 5;
+const DEFAULT_SORT = {
+  direction: 'desc' as const,
+  field: 'impact' as const,
+};
+
+const TransactionGroupLinkWrapper = styled.div`
+  width: 100%;
+  .euiToolTipAnchor {
+    width: 100% !important;
+  }
+`;
+
+const StyledTransactionDetailLink = styled(TransactionDetailLink)`
+  display: block;
+  ${truncate('100%')}
+`;
+
+export function ServiceOverviewTransactionsTable(props: Props) {
+  const { serviceName } = props;
+
+  const {
+    uiFilters,
+    urlParams: { start, end },
+  } = useUrlParams();
+
+  const [tableOptions, setTableOptions] = useState<{
+    pageIndex: number;
+    sort: {
+      direction: SortDirection;
+      field: SortField;
+    };
+  }>({
+    pageIndex: 0,
+    sort: DEFAULT_SORT,
+  });
+
+  const {
+    data = {
+      totalItemCount: 0,
+      items: [],
+      tableOptions: {
+        pageIndex: 0,
+        sort: DEFAULT_SORT,
+      },
+    },
+    status,
+  } = useFetcher(() => {
+    if (!start || !end) {
+      return;
+    }
+
+    return callApmApi({
+      pathname: '/api/apm/services/{serviceName}/overview_transaction_groups',
+      params: {
+        path: { serviceName },
+        query: {
+          start,
+          end,
+          uiFilters: JSON.stringify(uiFilters),
+          size: PAGE_SIZE,
+          numBuckets: 20,
+          pageIndex: tableOptions.pageIndex,
+          sortField: tableOptions.sort.field,
+          sortDirection: tableOptions.sort.direction,
+        },
+      },
+    }).then((response) => {
+      return {
+        items: response.transaction_groups,
+        totalItemCount: response.total_transaction_groups,
+        tableOptions: {
+          pageIndex: tableOptions.pageIndex,
+          sort: {
+            field: tableOptions.sort.field,
+            direction: tableOptions.sort.direction,
+          },
+        },
+      };
+    });
+  }, [
+    serviceName,
+    start,
+    end,
+    uiFilters,
+    tableOptions.pageIndex,
+    tableOptions.sort.field,
+    tableOptions.sort.direction,
+  ]);
+
+  const {
+    items,
+    totalItemCount,
+    tableOptions: { pageIndex, sort },
+  } = data;
+
+  const columns: Array<EuiBasicTableColumn<ServiceTransactionGroupItem>> = [
```

Comment:
I think we should prefer to use the `APIReturnType` helper over `PromiseReturnType` since the type of the actual api response.